ruddock/modules/users/helpers.py

```diff
@@ -52,17 +81,38 @@ def get_user_info(username):
     FROM members
       NATURAL JOIN members_extra
       NATURAL JOIN membership_types
-      NATURAL JOIN users
-    WHERE username=:u
+    WHERE user_id=:u
     """)
-  return flask.g.db.execute(query, u=username).first()
+  return flask.g.db.execute(query, u=user_id).first()
 
-def get_office_info(username):
-  """Procedure to get a user's officer info."""
+def remove_member(user_id):
+  """
+  Permanently removes a member from the members database, removing any
+  record of them from the website. Returns True on success and False otherwise.
+  Be careful with this!
+  """
   query = sqlalchemy.text("""
-    SELECT office_name, start_date, end_date
-    FROM office_assignments NATURAL JOIN users NATURAL JOIN offices
-    WHERE username = :u
-    ORDER BY start_date, end_date, office_name
-  """)
-  return flask.g.db.execute(query, u=username).fetchall()
+    DELETE
+    FROM members
+    WHERE user_id=:u
+    """)
+  # Return whether the query was successful or not. Since we're deleting
+  # on a primary key, we know that we only have to check for row count
+  # 1 or 0.
+  return flask.g.db.execute(query, u=user_id).rowcount == 1
+
+def edit_member_column(user_id, column, new_value):
+  """Edits a member using the new value given."""
+  # Using a string format instead of the sqlalchemy.text formatting method
+  # for the column name, because otherwise it will always have single quotes
+  # around it, which SQL doesn't like. The only more consistent way I can
+  # think of to do this is to use SQLAlchemy's built-in query objects instead
+  # of writing all the queries manually, but that involves changing a lot of
+  # existing code.
+  query = sqlalchemy.text("""
+    UPDATE members
+    SET {}=:new_val
+    WHERE user_id=:u
+    """.format(column))
```

Comment:
this is a big SQL injection risk! you can't trust _anything_ that comes from a POST request. possible solution is:

make a list of (column name, display name, type) called MEMBER_EDIT_FIELDS or something, and scope it so this whole file can see it. you can use this to:

1. generate the html template for the "edit a member" page
2. iterate through that list to do the edits to the table. since you've hard-coded it in this file, each of those keys should be safe to pass to .format().
3. depending on how you handle types, it might make validation simpler. might not. kind of a nice surprise if it does though.